packages/snowpack/src/commands/build.ts

```diff
@@ -147,55 +146,71 @@ export async function command(commandOptions: CommandOptions) {
 
   for (const [dirDisk, dirDest, allFiles] of includeFileSets) {
     for (const locOnDisk of allFiles) {
-      const {baseExt: fileExt} = getExt(locOnDisk);
-      let outLoc = locOnDisk.replace(dirDisk, dirDest);
-      let builtLocOnDisk = locOnDisk;
-      const extToReplace = config._extensionMap[fileExt] || srcFileExtensionMapping[fileExt];
-      if (extToReplace) {
-        outLoc = replaceExt(outLoc, extToReplace);
-        builtLocOnDisk = replaceExt(builtLocOnDisk, extToReplace);
-      }
+      const srcExt = path.extname(locOnDisk);
       const builtFileOutput = await buildFile(locOnDisk, {
         plugins: config.plugins,
         messageBus,
         isDev: false,
         isHmrEnabled: false,
       });
       allBuiltFromFiles.add(locOnDisk);
-      const {baseExt, expandedExt} = getExt(outLoc);
-      let contents =
-        builtFileOutput[
-          config._extensionMap[fileExt] || srcFileExtensionMapping[fileExt] || fileExt
-        ];
-      if (!contents) {
-        continue;
-      }
-      const cssOutPath = outLoc.replace(/.js$/, '.css');
-      mkdirp.sync(path.dirname(outLoc));
-      switch (baseExt) {
-        case '.js': {
-          if (builtFileOutput['.css']) {
-            await fs.mkdir(path.dirname(cssOutPath), {recursive: true});
-            await fs.writeFile(cssOutPath, builtFileOutput['.css'], 'utf-8');
-            contents = `import './${path.basename(cssOutPath)}';\n` + contents;
-          }
-          contents = wrapImportMeta({code: contents, env: true, isDev: false, hmr: false, config});
-          allFilesToResolveImports[outLoc] = {baseExt, expandedExt, contents, locOnDisk};
-          break;
+
+      for (const [fileExt, buildResult] of Object.entries(builtFileOutput)) {
+        let {code, map} = buildResult;
+        if (!code) {
+          continue;
         }
-        case '.html': {
-          contents = wrapHtmlResponse({
-            code: contents,
-            isDev: false,
 
-            hmr: false,
-            config,
-          });
-          allFilesToResolveImports[outLoc] = {baseExt, expandedExt, contents, locOnDisk};
-          break;
+        const outDir = path.dirname(locOnDisk.replace(dirDisk, dirDest));
+        const outFilename = replaceExt(path.basename(locOnDisk), srcExt, fileExt);
+        const outLoc = path.join(outDir, outFilename);
+        const sourceMappingURL = outFilename + '.map';
+
+        switch (fileExt) {
+          case '.css': {
+            if (map) code = cssSourceMappingURL(code, sourceMappingURL);
```

Comment:
here and others: if `sourceMap === false`, I wouldn't expect any source maps even if the plugin returned one. 

This would simplify things for plugin authors as well, where they can always return `map` if they'd like and then let Snowpack handle whether it gets written or not. Esp. for plugins like Svelte where the `map` is always returned.

Reply:
Let me handle that in the build file step, then, that way it’s only in one place.

Reply:
Updated in `build-pipeline.ts`